### ISSUE TITLE
Bug in loss recording for LAMA

### ISSUE THREAD

**Issue by lxuechen** [THUDM/P-tuning#20]
The `loss` variable recorded for the LAMA task isn't the average over the dataset. 

Since `loss` is always reset in the if-else block, the recorded loss on [this](https://github.com/THUDM/P-tuning/blob/368ab8561bab04b44010744a365124efaed6bf16/LAMA/cli.py#L136) line is always the double of the last batch. 

Pasting the relevant block here for reference
```python
for x_hs, x_ts in loader:
    if False and self.args.extend_data:
        loss, _hit1 = self.model.test_extend_data(x_hs, x_ts)
    elif evaluate_type == 'Test':
        loss, _hit1, top10 = self.model(x_hs, x_ts, return_candidates=True)
    else:
        loss, _hit1 = self.model(x_hs, x_ts)
    hit1 += _hit1
    loss += loss.item()
```

**Comment by Xiao9905** [comment]
Thanks for your reminding. We will fix it in our next major update.
